### PR TITLE
Fixed deprecations

### PR DESCRIPTION
**Pull request by BillyD73**

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -81,6 +81,7 @@ dependencies {
     compile group:'com.fasterxml.jackson.core', name:'jackson-core', version:jackson_version
     compile group:'com.fasterxml.jackson.core', name:'jackson-databind', version:jackson_version
     compile group:'com.fasterxml.jackson.core', name:'jackson-annotations', version:jackson_version
+    compile group:'org.jmockit', name:'jmockit', version:jmockit_version
```
Comment (Azquelt) [inline server/build.gradle]: Why is this change needed?

Won't this result in jmockit being included in the .war?

File: server/src/main/java/com/ibm/ws/lars/rest/mongo/PersistenceBean.java [hunk 2]
```diff
@@ -87,6 +88,7 @@
     private static final String ID = "_id";
 
     @Inject
+    @Injectable
     private com.mongodb.DB db;
```
Comment (Azquelt) [inline server/src/main/java/com/ibm/ws/lars/rest/mongo/PersistenceBean.java]: Are you _sure_ this is needed here? It's really odd to have a test framework annotation in your production code.

Comment (BillyD73) [reply server/src/main/java/com/ibm/ws/lars/rest/mongo/PersistenceBean.java]: I was led to believe this was needed to allow for the reflection fix to Deencapsulation.setField() being deprecated - will push an updated PR with this removed, thanks Andy!

File: client-lib-tests/src/fat/java/com/ibm/ws/repository/test/RepositoryUtilsTest.java [hunk 3]
```diff
@@ -77,13 +77,13 @@ public void testRepositoryStatusOK() throws IOException, RequestFailureException
...
     public void testInvalidRepositoryNotAvailable() {
-        RestRepositoryConnection invalidLoginInfo = new RestRepositoryConnection("I", "don't", "exist", "http://dont.exist.com");
+        RestRepositoryConnection invalidLoginInfo = new RestRepositoryConnection("I", "don't", "exist", "http://dont252qmadjf842sgs7842d.exist.com");
         assertFalse("An invalid test repository should not be available", invalidLoginInfo.isRepositoryAvailable());
     }
 
     @Test
     public void testInvalidRepositoryThrowsException() throws IOException, RequestFailureException {
-        RestRepositoryConnection invalidLoginInfo = new RestRepositoryConnection("I", "don't", "exist", "http://dont.exist.com");
+        RestRepositoryConnection invalidLoginInfo = new RestRepositoryConnection("I", "don't", "exist", "http://dont252qmadjf842sgs7842d.exist.com");
```
Comment (rickettmwork) [inline client-lib-tests/src/fat/java/com/ibm/ws/repository/test/RepositoryUtilsTest.java]: This will still hit exist.com's DNS, which probably isn't terribly polite
I'd suggest since we're changing it, we should change it to doesntexist.ibm.com

**Review by rickettmwork — CHANGES_REQUESTED** [review]

**Review by pnickoll — APPROVED** [review]

File: client-lib-tests/src/fat/java/com/ibm/ws/repository/test/RepositoryUtilsTest.java (hunk 3 above)
Comment (BillyD73) [reply client-lib-tests/src/fat/java/com/ibm/ws/repository/test/RepositoryUtilsTest.java]: Ahhh of course! Will fix that and update. Thank you :)

**Review by rickettmwork — APPROVED** [review]

**Review by Azquelt — APPROVED** [review]
